lib/SILOptimizer/Mandatory/MandatoryInlining.cpp

```diff
@@ -433,20 +433,16 @@ static std::tuple<FullApplySite, SILBasicBlock::iterator>
 tryDevirtualizeApplyHelper(FullApplySite InnerAI, SILBasicBlock::iterator I,
                            ClassHierarchyAnalysis *CHA) {
   auto NewInstPair = tryDevirtualizeApply(InnerAI, CHA);
-  auto *NewInst = NewInstPair.first;
-  if (!NewInst)
+  if (!NewInstPair.second) {
     return std::make_tuple(InnerAI, I);
+  }
 
-  replaceDeadApply(InnerAI, NewInst);
+  replaceDeadApply(InnerAI, NewInstPair.first);
 
   auto newApplyAI = NewInstPair.second.getInstruction();
   assert(newApplyAI && "devirtualized but removed apply site?");
   I = newApplyAI->getIterator();
   auto NewAI = FullApplySite::isa(newApplyAI);
```

Comment:
Why not just eliminate NewAI, i.e.:

```
return {FullApplySite::isa(newApplyAI), I};
```

Also next time, it would be helpful for reviewers who read the mailing list if you could put stuff like this in a separate focused PR. The reason why I am asking is that logically has nothing to do with the title/subject of the PR, then people who just look at Titles on the mailing list may not know that this is code that they should review.

Reply:
Done.

Reply:
And yes logically the SIL optimizer changes are distinct, and could be in a separate PR, but they were motivated by other changes in this PR. It can be tricky to find the right balance here. Sometimes, splitting up a change into too many PRs can also be difficult to review because its hard to get the big picture.

Reply:
That is why in the PR you specifically tell the reviewer the big picture and uses references in between the PRs/etc to provide the connection for posterity. *NOTE* I am not saying that I have always done this well.